python/tvm/contrib/tedd.py

```diff
@@ -0,0 +1,506 @@
+        get_or_create_dot_id.obj_id_dict = {}
+    if obj not in get_or_create_dot_id.obj_id_dict:
+        if assert_on_missing:
+            assert False, 'dot_id ' + str(obj) + ' has not been registered.'
+        else:
+            get_or_create_dot_id.obj_id_dict[obj] = prefix + hex(id(obj))
+    return get_or_create_dot_id.obj_id_dict[obj]
+
+
+def get_port_id(is_input, index):
+    return 'I_' + str(index) if is_input else 'O_' + str(index)
+
+
+def get_itervar_type_info(iter_type):
+    assert iter_type < len(
+        ITERVAR_TYPE_STRING_MAP), 'Unknown IterVar type: ' + str(iter_type)
+    return ITERVAR_TYPE_STRING_MAP[iter_type]
+
+
+def get_itervar_label_color(itervar, iv_type):
+    type_info = get_itervar_type_info(iv_type)
+    return str(itervar.var) + '(' + type_info[0] + ')', type_info[1]
+
+
+def linebrk(s, n):
+    """ Break input string s with <br/> for every n charactors."""
+    result = ''
+    j = 0
+    for i, c in enumerate(s):
+        if j == n and i != len(s) - 1:
+            result = result + '\n'
+            j = 0
+        j = j + 1
+        result = result + c
+    result = html.escape(str(result), quote=True)
+    result = result.replace('\n', '<br/>')
+    return result
+
+
+def create_graph(name="", rankdir='BT'):
+    graph = Digraph(name=name)
+    graph.graph_attr['rankdir'] = rankdir
+    return graph
+
+
+def itervar_label(itervar, index, index_color, label):
+    return '<TR><TD PORT="' + get_or_create_dot_id(itervar, str(
+        itervar.var)) + '" BGCOLOR="' + index_color + '">' + str(
+            index
+        ) + '</TD><TD BGCOLOR="white" PORT="itervar">' + label + '</TD></TR>'
+
+
+def stage_label(stage):
+    return stage.op.name + '<br/>Scope: ' + stage.scope
+
+
+def legend_label():
+    label = '<<TABLE BORDER="0" CELLBORDER="1" CELLSPACING="0" CELLPADDING="4">'
+    for iter_type in ITERVAR_TYPE_STRING_MAP:
+        name, color = ITERVAR_TYPE_STRING_MAP[iter_type]
+        label += '<TR><TD BGCOLOR="' + color + '"></TD>' \
+                + '<TD BGCOLOR="white">' + name + '</TD></TR>'
+    label += '</TABLE>>'
+    return label
+
+
+def legend_dot(g):
+    with g.subgraph(name='cluster_legend') as subgraph:
+        subgraph.attr(label='Legend')
+        label = legend_label()
+        subgraph.node('legend', label, shape='none', margin='0')
+
+
+def dump_graph(dot_string,
+               showsvg=True,
+               dotfilepath='',
+               outputdotstring=False):
+    """Output dot_string in various formats."""
+    if dotfilepath:
+        try:
+            dot_file = open(dotfilepath, "w+")
+            dot_file.write(dot_string)
+            dot_file.close()
+        except IOError:
+            print('Cannot open file: ' + dotfilepath)
+    if showsvg:
+        src = Source(dot_string)
+        display(SVG(src.pipe(format='svg')))
+    if outputdotstring:
+        return dot_string
+    return None
+
+
+def viz_schedule_tree(sch,
+                      showsvg=False,
```

Comment:
Can you please change the argument name to `show_svg`, `dot_file_path` and `output_dot_string`?

Reply:
I am making these changes.

Reply:
@Hzfengsy, I have updated the variable names and avoided some lint warnings from the tests.  But the test will still fail because of the ci-cpu doesn't have graphviz.  I haven't seen response to my question below.  Do you have any suggestion?  Thanks

Reply:
I am also working on a tutorial.  How about putting it under tutorial/language?

Reply:
Sounds great! As for the CI test, I would like to skip this test on the CI until the CI updating.

Also, is there a way to allow contributors to modify the docker image on CI? @tqchen 
Many new features depend on different environments. It is difficult to change the CI docker every time by only one person.

Reply:
I have disabled TEDD execution and verification and kept TE construction in the test.